--- a/com/pixelmongenerations/common/spawning/spawners/SpawnerTimed.java
+++ b/com/pixelmongenerations/common/spawning/spawners/SpawnerTimed.java
@@ -61,7 +61,7 @@
                     if (this.isBlockValidForPixelmonSpawning(world, new BlockPos(xPos, topEarth, zPos))) {
                         cpY = topEarth;
                     } else {
-                        String biomeID = world.getBiome(pos).getRegistryName().getPath();
+                        String biomeID = world.getBiome(pos).getRegistryName().getResourcePath();
                         if (PixelmonBiomeDictionary.isBiomeOfType(biomeID, EnumBiomeType.JUNGLE)) {
                             for(int i = 0; i < 5; ++i) {
                                 if (this.isBlockValidForPixelmonSpawning(world, new BlockPos(xPos, topEarth + i, zPos))) {
@@ -144,7 +144,7 @@
         validFloorMaterials.add(Material.SNOW);
         validFloorMaterials.add(Material.PACKED_ICE);
         validFloorMaterials.add(Material.PLANTS);
-        validAirMaterials = new HashSet();
+        validAirMaterials = new HashSet<>();
         validAirMaterials.add(Material.AIR);
         validAirMaterials.add(Material.WATER);
     }
