--- a/com/pixelmongenerations/common/spawning/spawners/SpawnerTimed.java
+++ b/com/pixelmongenerations/common/spawning/spawners/SpawnerTimed.java
@@ -144,7 +144,7 @@
         validFloorMaterials.add(Material.SNOW);
         validFloorMaterials.add(Material.PACKED_ICE);
         validFloorMaterials.add(Material.PLANTS);
-        validAirMaterials = new HashSet();
+        validAirMaterials = new HashSet<>();
         validAirMaterials.add(Material.AIR);
         validAirMaterials.add(Material.WATER);
     }
