--- a/com/pixelmongenerations/common/block/BlockPicketFence.java
+++ b/com/pixelmongenerations/common/block/BlockPicketFence.java
@@ -48,7 +48,7 @@
         super(Material.WOOD);
         this.setHardness(1.0F);
         this.setSoundType(SoundType.WOOD);
-        this.setTranslationKey("picket_fence_normal");
+        this.setTranslationKey("picket_fence_normal");
     }
 
     public void addCollisionBoxToList(IBlockState state, World worldIn, BlockPos pos, AxisAlignedBB entityBox, List<AxisAlignedBB> collidingBoxes, @Nullable Entity entityIn) {
