--- a/com/pixelmongenerations/common/block/BlockLeafEvolutionOre.java
+++ b/com/pixelmongenerations/common/block/BlockLeafEvolutionOre.java
@@ -25,7 +25,7 @@
         this.setSoundType(SoundType.STONE);
         this.setCreativeTab(PixelmonCreativeTabs.natural);
         this.setTickRandomly(true);
-        this.setTranslationKey(itemName);
+        this.setUnlocalizedName(itemName);
     }
 
     public boolean isOpaqueCube(IBlockState state) {
