--- a/com/pixelmongenerations/common/block/MultiBlock.java
+++ b/com/pixelmongenerations/common/block/MultiBlock.java
@@ -37,7 +37,7 @@
     public int width;
     public int length;
     public double height;
-    private static HashMap<Block, BoundingBoxSet> boundingBoxes;
+    private static final HashMap<Block, BoundingBoxSet> boundingBoxes;
 
     protected MultiBlock(Material material, int width, double height, int length) {
         super(material);
@@ -47,7 +47,7 @@
     }
 
     protected BlockStateContainer createBlockState() {
-        return new BlockStateContainer(this, new IProperty[]{FACING, MULTIPOS});
+        return new BlockStateContainer(this, FACING, MULTIPOS);
     }
 
     public IBlockState getStateFromMeta(int meta) {
@@ -141,7 +141,7 @@
     }
 
     public TileEntity createTileEntity(World world, IBlockState state) {
-        return state.getValue(MULTIPOS) == EnumMultiPos.BASE ? (TileEntity)this.getTileEntity(world, state).orElse((Object)null) : null;
+        return state.getValue(MULTIPOS) == EnumMultiPos.BASE ? (TileEntity)this.getTileEntity(world, state).orElse(null) : null;
     }
 
     protected abstract Optional<TileEntity> getTileEntity(World var1, IBlockState var2);
