--- a/com/pixelmongenerations/common/block/PixelmonBlockFalling.java
+++ b/com/pixelmongenerations/common/block/PixelmonBlockFalling.java
@@ -44,7 +44,6 @@
 
     private void checkFallable(World worldIn, BlockPos pos) {
         if ((worldIn.isAirBlock(pos.down()) || canFallThrough(worldIn.getBlockState(pos.down()))) && pos.getY() >= 0) {
-            int i = true;
             if (!fallInstantly && worldIn.isAreaLoaded(pos.add(-32, -32, -32), pos.add(32, 32, 32))) {
                 if (!worldIn.isRemote) {
                     EntityFallingBlock entityfallingblock = new EntityFallingBlock(worldIn, (double)pos.getX() + 0.5D, (double)pos.getY(), (double)pos.getZ() + 0.5D, worldIn.getBlockState(pos));
