--- a/com/pixelmongenerations/common/block/PixelmonLog.java
+++ b/com/pixelmongenerations/common/block/PixelmonLog.java
@@ -14,7 +14,7 @@
 
 public class PixelmonLog extends BlockLog {
     public PixelmonLog(String name) {
-        this.setTranslationKey(name);
+        this.setTranslationKey(name);
         this.setSoundType(SoundType.WOOD);
         this.setDefaultState(this.blockState.getBaseState().withProperty(LOG_AXIS, EnumAxis.Y));
         this.setCreativeTab(PixelmonCreativeTabs.buildingBlocks);
