--- a/com/pixelmongenerations/api/spawning/conditions/SpawnCondition.java
+++ b/com/pixelmongenerations/api/spawning/conditions/SpawnCondition.java
@@ -113,12 +113,9 @@
 
     public boolean fits(SpawnInfo spawnInfo, SpawnLocation spawnLocation) {
         int ticks = (int)(spawnLocation.location.world.getWorldTime() % 24000L);
-        boolean timeFits = false;
+        boolean timeFits;
         if (this.times != null && !this.times.isEmpty()) {
-            Stream var10000 = WorldTime.getCurrent(ticks).stream();
-            ArrayList var10001 = this.times;
-            var10001.getClass();
-            timeFits = var10000.anyMatch(var10001::contains);
+            timeFits = WorldTime.getCurrent(ticks).stream().anyMatch(this.times::contains);
         } else {
             timeFits = true;
         }
