--- a/com/pixelmongenerations/client/gui/factory/config/ConfigurateNodeElement.java
+++ b/com/pixelmongenerations/client/gui/factory/config/ConfigurateNodeElement.java
@@ -3,8 +3,8 @@
 import com.google.common.collect.Lists;
 import com.pixelmongenerations.core.Pixelmon;
 import com.pixelmongenerations.core.config.PixelmonConfig;
-import info.pixelmon.shadow.ninja.leaping.configurate.ConfigurationNode;
-import info.pixelmon.shadow.ninja.leaping.configurate.commented.CommentedConfigurationNode;
+import ninja.leaping.configurate.ConfigurationNode;
+import ninja.leaping.configurate.commented.CommentedConfigurationNode;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Arrays;
