--- a/com/pixelmongenerations/common/entity/npcs/EntityNPC.java
+++ b/com/pixelmongenerations/common/entity/npcs/EntityNPC.java
@@ -356,9 +356,9 @@
     public static <T extends EntityNPC> Optional<T> locateNPCClient(World world, int id, Class<T> type) {
         try {
             for(int i = 0; i < world.loadedEntityList.size(); ++i) {
-                Entity entity = (Entity)world.loadedEntityList.get(i);
+                Entity entity = world.loadedEntityList.get(i);
                 if (type.isInstance(entity) && ((EntityNPC)entity).getId() == id) {
-                    return Optional.of((EntityNPC)entity);
+                    return (Optional<T>) Optional.of((EntityNPC)entity);
                 }
             }
 
