--- a/com/pixelmongenerations/common/block/spawning/BlockCustomPixelmonGrass.java
+++ b/com/pixelmongenerations/common/block/spawning/BlockCustomPixelmonGrass.java
@@ -49,7 +49,7 @@
         this.setBlockUnbreakable();
         this.setResistance(6000000.0F);
         this.setCreativeTab(PixelmonCreativeTabs.utilityBlocks);
-        this.setTranslationKey("pixelmonCustomGrass");
+        this.setUnlocalizedName("pixelmonCustomGrass");
     }
 
     public void onEntityWalk(World worldIn, BlockPos pos, Entity entityIn) {
