--- a/com/pixelmongenerations/api/events/PokeballEffectEvent.java
+++ b/com/pixelmongenerations/api/events/PokeballEffectEvent.java
@@ -18,21 +18,21 @@
     @Cancelable
     public static class SentOutEffect extends PokeballEffectEvent {
         public SentOutEffect(EntityPokeBall pokeBall) {
-            super(pokeBall, null);
+            super(pokeBall);
         }
     }
 
     @Cancelable
     public static class SuccessfullCaptureEffect extends PokeballEffectEvent {
         public SuccessfullCaptureEffect(EntityPokeBall pokeBall) {
-            super(pokeBall, null);
+            super(pokeBall);
         }
     }
 
     @Cancelable
     public static class StartCaptureEffect extends PokeballEffectEvent {
         public StartCaptureEffect(EntityPokeBall pokeBall) {
-            super(pokeBall, null);
+            super(pokeBall);
         }
     }
 }
