--- a/com/pixelmongenerations/common/block/machines/BlockVendingMachine.java
+++ b/com/pixelmongenerations/common/block/machines/BlockVendingMachine.java
@@ -30,7 +30,7 @@
         super(Material.IRON, 1, 2.0D, 1);
         this.setHardness(2.0F);
         this.setSoundType(SoundType.METAL);
-        this.setTranslationKey(color + "VendingMachine");
+        this.setUnlocalizedName(color + "VendingMachine");
         this.color = color;
     }
 
