--- a/com/pixelmongenerations/common/entity/pixelmon/abilities/Drizzle.java
+++ b/com/pixelmongenerations/common/entity/pixelmon/abilities/Drizzle.java
@@ -10,13 +10,9 @@
 public class Drizzle extends AbilityBase {
     public void applySwitchInEffect(PixelmonWrapper newPokemon) {
         if (newPokemon.getSpecies() != EnumSpecies.Kyogre || newPokemon.getHeldItem().getHeldItemType() != EnumHeldItems.blueOrb) {
-            Stream var10000 = newPokemon.bc.getActivePokemon().stream().map(PixelmonWrapper::getBattleAbility);
-            WeatherTrio.class.getClass();
+            Stream<AbilityBase> var10000 = newPokemon.bc.getActivePokemon().stream().map(PixelmonWrapper::getBattleAbility);
             var10000 = var10000.filter(WeatherTrio.class::isInstance);
-            WeatherTrio.class.getClass();
-            if (var10000.map(WeatherTrio.class::cast).map((a) -> {
-                return a.weather;
-            }).noneMatch(Weather::isWeatherTrioStatus) && !(newPokemon.bc.globalStatusController.getWeatherIgnoreAbility() instanceof Rainy)) {
+            if (var10000.map(WeatherTrio.class::cast).map((a) -> a.weather).noneMatch(Weather::isWeatherTrioStatus) && !(newPokemon.bc.globalStatusController.getWeatherIgnoreAbility() instanceof Rainy)) {
                 Rainy rainy = new Rainy();
                 rainy.setStartTurns(newPokemon);
                 newPokemon.bc.globalStatusController.addGlobalStatus(rainy);
