--- a/com/pixelmongenerations/common/entity/pixelmon/abilities/Reckless.java
+++ b/com/pixelmongenerations/common/entity/pixelmon/abilities/Reckless.java
@@ -7,7 +7,7 @@
 public class Reckless extends AbilityBase {
     public int[] modifyPowerAndAccuracyUser(int power, int accuracy, PixelmonWrapper user, PixelmonWrapper target, Attack a) {
         String[] recoilMoves = new String[]{"Brave Bird", "Double-Edge", "Flare Blitz", "Head Charge", "Head Smash", "Hi Jump Kick", "Jump Kick", "Submission", "Take Down", "Volt Tackle", "Wild Charge", "Wood Hammer"};
-        if (Arrays.asList(recoilMoves).contains(a.getAttackBase().getUnlocalizedName())) {
+        if (Arrays.asList(recoilMoves).contains(a.getAttackBase().getTranslationKey())) {
             power = (int)((double)power * 1.2D);
         }
 
