--- a/com/pixelmongenerations/common/block/apricornTrees/BlockApricornTree.java
+++ b/com/pixelmongenerations/common/block/apricornTrees/BlockApricornTree.java
@@ -54,7 +54,7 @@
         this.tree = tree;
         this.setTickRandomly(true);
         this.setHardness(2.0F);
-        this.setTranslationKey("apricorn tree");
+        this.setUnlocalizedName("apricorn tree");
         this.setRegistryName("pixelmon:" + tree.name().toLowerCase() + "_apricorn_tree");
     }
 
