--- a/com/pixelmongenerations/client/models/tmt/impl/ModelRendererTurbo.java
+++ b/com/pixelmongenerations/client/models/tmt/impl/ModelRendererTurbo.java
@@ -621,7 +621,6 @@
         f4 += scale;
         f5 += scale;
         f6 += scale;
-        int m = this.mirror ? true : true;
         if (this.mirror) {
             float f7 = f4;
             f4 = x;
@@ -770,9 +769,6 @@
         float x1 = x - expansion;
         float y1 = y - expansion;
         float z1 = z - expansion;
-        int wDir = false;
-        int hDir = false;
-        int dDir = false;
         float wScale = 1.0F + expansion / ((float)w * pixelScale);
         float hScale = 1.0F + expansion / ((float)h * pixelScale);
         byte wDir;
@@ -871,7 +867,6 @@
                 int curVert = 1 + i + segs * (j - 1);
                 tempVerts[curVert] = new PositionTextureVertex(x + xSize * r, y + yHeight * r, z + zSize * r, 0.0F, 0.0F);
                 if (i > 0) {
-                    PositionTextureVertex[] verts;
                     if (j == 1) {
                         verts = new PositionTextureVertex[]{tempVerts[curVert].setTexturePosition(startU + segW * (float)i, startV + segH * (float)j), tempVerts[curVert - 1].setTexturePosition(startU + segW * (float)(i - 1), startV + segH * (float)j), tempVerts[0].setTexturePosition(startU + segW * (float)(i - 1), startV), tempVerts[0].setTexturePosition(startU + segW + segW * (float)i, startV)};
                     } else {
