--- a/mcmod.info
+++ b/mcmod.info
@@ -5,11 +5,10 @@
   "description": "Pokemon in Minecraft!",
   "version": "8.2.2",
   "mcversion": "1.12.2",
-  "forgeversion": "Forge 14.23.5.2768",
   "url": "https://pixelmongenerations.com/",
   "updateUrl": "https://pixelmongenerations.com/",
-  "authorList": ["Pixelmon Generations Development Team"],
-  "credits": "Our community, our team, and the original Pixelmon team.",
+  "authorList": ["They dont deserve credit"],
+  "credits": "Our community and the original Pixelmon team.",
   "logoFile": "",
   "screenshots": [],
   "dependencies": []
