--- a/com/pixelmongenerations/common/entity/pixelmon/stats/ExtraStats.java
+++ b/com/pixelmongenerations/common/entity/pixelmon/stats/ExtraStats.java
@@ -19,10 +19,8 @@
     public abstract void readFromNBT(NBTTagCompound var1);
 
     public static ExtraStats getExtraStats(String name) {
-        Optional var10000 = EnumSpecies.getFromName(name);
-        EnumMap var10001 = statsList;
-        var10001.getClass();
-        return (ExtraStats)var10000.map(var10001::get).map(Supplier::get).orElse((Object)null);
+        Optional<EnumSpecies> var10000 = EnumSpecies.getFromName(name);
+        return var10000.map(statsList::get).map(Supplier::get).orElse(null);
     }
 
     public static void getNBTTags(HashMap<String, Class> tags) {
