--- a/com/pixelmongenerations/common/block/multiBlocks/BlockFridge.java
+++ b/com/pixelmongenerations/common/block/multiBlocks/BlockFridge.java
@@ -25,7 +25,7 @@
         super(Material.ROCK, 1, 2.0D, 1);
         this.setHardness(0.5F);
         this.setSoundType(SoundType.WOOD);
-        this.setTranslationKey("Fridge");
+        this.setUnlocalizedName("Fridge");
     }
 
     public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, EntityPlayer player, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ) {
