--- a/com/pixelmongenerations/common/block/multiBlocks/BlockFossilDisplay.java
+++ b/com/pixelmongenerations/common/block/multiBlocks/BlockFossilDisplay.java
@@ -32,14 +32,14 @@
     public BlockFossilDisplay() {
         super(Material.ROCK, 1, 2.0D, 1);
         this.color = null;
-        this.setTranslationKey("FossilDisplay");
+        this.setTranslationKey("FossilDisplay");
         this.setHardness(2.0F);
     }
 
     public BlockFossilDisplay(ColorEnum colorEnum) {
         this();
         this.color = colorEnum;
-        this.setTranslationKey(this.color + "FossilDisplay");
+        this.setTranslationKey(this.color + "FossilDisplay");
     }
 
     public List<ItemStack> getDrops(IBlockAccess world, BlockPos pos, IBlockState state, int fortune) {
