--- a/com/pixelmongenerations/common/block/spawnmethod/BlockTaoTrioShrine.java
+++ b/com/pixelmongenerations/common/block/spawnmethod/BlockTaoTrioShrine.java
@@ -11,7 +11,7 @@
 
 public class BlockTaoTrioShrine extends BlockShrine {
     public BlockTaoTrioShrine() {
-        this.setTranslationKey("tao_trio_shrine");
+        this.setTranslationKey("tao_trio_shrine");
         this.setBlockUnbreakable();
     }
 
