--- a/com/pixelmongenerations/client/gui/pokemoneditor/GuiIndividualEditorBase.java
+++ b/com/pixelmongenerations/client/gui/pokemoneditor/GuiIndividualEditorBase.java
@@ -35,9 +35,9 @@
     private GuiTextField tbName;
     private GuiTextField tbNickname;
     private GuiTextField tbLvl;
-    private GuiTextField[] tbMoves = new GuiTextField[4];
+    private final GuiTextField[] tbMoves = new GuiTextField[4];
     private String origName;
-    private Attack[] attacks = new Attack[4];
+    private final Attack[] attacks = new Attack[4];
     protected GuiTextField formText = null;
     private static final int BUTTON_OKAY = 1;
     private static final int BUTTON_GENDER = 3;
@@ -59,13 +59,13 @@
         this.buttonList.clear();
         this.buttonList.add(new GuiButton(1, this.width / 2 + 155, this.height / 2 + 90, 30, 20, I18n.translateToLocal("gui.guiItemDrops.ok")));
         this.updateTextureDropDown();
-        int malePercent = ((BaseStats)Entity3HasStats.getBaseStats(this.p.name).get()).malePercent;
+        int malePercent = Entity3HasStats.getBaseStats(this.p.name).get().malePercent;
         if (malePercent > 0 && malePercent < 100) {
             this.buttonList.add(new GuiButtonHoverDisable(3, this.width / 2 - 120, this.height / 2 + 30, 80, 20, this.p.gender == Gender.Male ? I18n.translateToLocal("gui.trainereditor.male") : I18n.translateToLocal("gui.trainereditor.female")));
         }
 
         this.addButton(new GuiButton(30, this.width / 2 + 15, this.height / 2 + 30, 30, 20, this.p.isShiny ? "Yes" : "No"));
-        this.addDropDown((new GuiDropDown(EnumGrowth.getProperOrder(), this.p.growth, this.width / 2 - 120, this.height / 2 + 28, 80, 100)).setOnSelected((growth) -> {
+        this.addDropDown((new GuiDropDown<>(EnumGrowth.getProperOrder(), this.p.growth, this.width / 2 - 120, this.height / 2 + 28, 80, 100)).setOnSelected((growth) -> {
             this.p.growth = growth;
         }).setGetOptionString(EnumGrowth::getLocalizedName).setInactiveTop(this.height / 2 + 54));
         this.tbName = new GuiTextField(6, this.mc.fontRenderer, this.width / 2 - 120, this.height / 2 - 50, 90, 17);
@@ -122,7 +122,7 @@
         }
 
         this.mc.renderEngine.bindTexture(GuiResources.cwPanel);
-        GuiHelper.drawImageQuad((double)(this.width / 2 - 200), (double)(this.height / 2 - 120), 400.0D, 240.0F, 0.0D, 0.0D, 1.0D, 1.0D, this.zLevel);
+        GuiHelper.drawImageQuad(this.width / 2 - 200, this.height / 2 - 120, 400.0D, 240.0F, 0.0D, 0.0D, 1.0D, 1.0D, this.zLevel);
         RenderHelper.disableStandardItemLighting();
         this.mc.fontRenderer.drawString(this.titleText, this.width / 2 - this.mc.fontRenderer.getStringWidth(this.titleText) / 2, this.height / 2 - 90, 0);
         this.mc.fontRenderer.drawString(I18n.translateToLocal("gui.trainereditor.name"), this.width / 2 - 180, this.height / 2 - 45, 0);
@@ -137,7 +137,7 @@
         this.mc.fontRenderer.drawString(I18n.translateToLocal("gui.trainereditor.growth"), this.width / 2 - 180, this.height / 2 + 55, 0);
         GuiHelper.bindPokemonSprite(this.p, this.mc);
         GlStateManager.color(1.0F, 1.0F, 1.0F);
-        GuiHelper.drawImageQuad((double)(this.width / 2 - 157), (double)(this.height / 2 - 73 - (this.p.isGen6Sprite() ? -3 : 0)), 20.0D, 20.0F, 0.0D, 0.0D, 1.0D, 1.0D, 1.0F);
+        GuiHelper.drawImageQuad(this.width / 2 - 157, this.height / 2 - 73 - (this.p.isGen6Sprite() ? -3 : 0), 20.0D, 20.0F, 0.0D, 0.0D, 1.0D, 1.0D, 1.0F);
         this.mc.fontRenderer.drawString(I18n.translateToLocal("gui.screenpokechecker.moves"), this.width / 2 + 100, this.height / 2 - 15, 0);
         GuiTextField[] var4 = this.tbMoves;
         int var5 = var4.length;
@@ -228,7 +228,7 @@
         }
 
         this.removeDropDown(this.texture);
-        this.texture = (new GuiDropDown(shinyOptions, cuurentType, this.width / 2 - 120, this.height / 2 + 14, 80, 50)).setOnSelected(this::setTextureType).setGetOptionString(IEnumSpecialTexture::name);
+        this.texture = (new GuiDropDown<>(shinyOptions, cuurentType, this.width / 2 - 120, this.height / 2 + 14, 80, 50)).setOnSelected(this::setTextureType).setGetOptionString(IEnumSpecialTexture::name);
         this.addDropDown(this.texture);
     }
 
@@ -237,10 +237,8 @@
         this.tbNickname.mouseClicked(x, y, button);
         this.tbName.mouseClicked(x, y, button);
         GuiTextField[] var4 = this.tbMoves;
-        int var5 = var4.length;
 
-        for(int var6 = 0; var6 < var5; ++var6) {
-            GuiTextField tbMove = var4[var6];
+        for (GuiTextField tbMove : var4) {
             tbMove.mouseClicked(x, y, button);
         }
 
@@ -255,14 +253,14 @@
         if (button.enabled) {
             if (button.id == 1) {
                 this.saveFields();
-            } else if (button.id == 3 && ((BaseStats)Entity3HasStats.getBaseStats(this.p.name).get()).malePercent > 0) {
+            } else if (button.id == 3 && Entity3HasStats.getBaseStats(this.p.name).get().malePercent > 0) {
                 this.p.gender = this.p.gender == Gender.Male ? Gender.Female : Gender.Male;
                 button.displayString = this.p.gender == Gender.Male ? I18n.translateToLocal("gui.trainereditor.male") : I18n.translateToLocal("gui.trainereditor.female");
             } else if (button.id == 4) {
                 Optional<EnumSpecies> pokemon = EnumSpecies.getFromName(this.tbName.getText());
                 if (pokemon.isPresent()) {
                     this.origName = this.tbName.getText();
-                    this.changePokemon((EnumSpecies)pokemon.get());
+                    this.changePokemon(pokemon.get());
                 } else {
                     this.tbName.setText(this.origName);
                 }
