--- a/com/pixelmongenerations/core/config/MigrationHandler.java
+++ b/com/pixelmongenerations/core/config/MigrationHandler.java
@@ -1,7 +1,7 @@
 package com.pixelmongenerations.core.config;
 
 import com.pixelmongenerations.core.Pixelmon;
-import info.pixelmon.shadow.ninja.leaping.configurate.commented.CommentedConfigurationNode;
+import ninja.leaping.configurate.commented.CommentedConfigurationNode;
 import java.util.HashMap;
 import java.util.function.Consumer;
 
