--- a/com/pixelmongenerations/core/config/PixelmonBlocks.java
+++ b/com/pixelmongenerations/core/config/PixelmonBlocks.java
@@ -1917,7 +1917,7 @@
         ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(ultraJungleLog), 0, new ModelResourceLocation("pixelmon:ultra_jungle_log", "inventory"));
         ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(ultraDarkLog), 0, new ModelResourceLocation("pixelmon:ultra_dark_log", "inventory"));
         ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(zCrystalOre), 0, new ModelResourceLocation("pixelmon:z_crystal_ore", "inventory"));
-        VariantRegister.registerVariantBlockItemModels(ultra.getDefaultState(), BlockColored.COLOR, EnumDyeColor::func_176765_a);
+        VariantRegister.registerVariantBlockItemModels(ultra.getDefaultState(), BlockColored.COLOR, EnumDyeColor::getMetadata);
         VariantRegister.registerVariantBlockItemModels(softSoil.getDefaultState(), BlockSoftSoil.MULCH, Enum::ordinal);
         VariantRegister.registerVariantBlockItemModels(ultraSandstone.getDefaultState(), BlockUltraSandStone.TYPE, BlockUltraSandStone.EnumType::getMetadata);
 
