--- a/com/pixelmongenerations/core/config/RegistryListener.java
+++ b/com/pixelmongenerations/core/config/RegistryListener.java
@@ -107,8 +107,8 @@
 
             for(int var5 = 0; var5 < var4; ++var5) {
                 Field field = var3[var5];
-                if (Modifier.isPublic(field.getModifiers()) && Modifier.isStatic(field.getModifiers()) && type.isInstance(field.get((Object)null))) {
-                    registry.register((IForgeRegistryEntry)field.get((Object)null));
+                if (Modifier.isPublic(field.getModifiers()) && Modifier.isStatic(field.getModifiers()) && type.isInstance(field.get(null))) {
+                    registry.register((IForgeRegistryEntry)field.get(null));
                 }
             }
         } catch (Exception var7) {
@@ -125,8 +125,8 @@
 
             for(int var3 = 0; var3 < var2; ++var3) {
                 Field field = var1[var3];
-                if (Modifier.isPublic(field.getModifiers()) && Modifier.isStatic(field.getModifiers()) && field.get((Object)null) instanceof Item) {
-                    Item item = (Item)field.get((Object)null);
+                if (Modifier.isPublic(field.getModifiers()) && Modifier.isStatic(field.getModifiers()) && field.get(null) instanceof Item) {
+                    Item item = (Item)field.get(null);
                     ModelLoader.setCustomModelResourceLocation(item, 0, new ModelResourceLocation(item.getRegistryName(), "inventory"));
                 }
             }
