--- a/com/pixelmongenerations/common/item/ItemShrineOrbWithInfo.java
+++ b/com/pixelmongenerations/common/item/ItemShrineOrbWithInfo.java
@@ -33,7 +33,7 @@
         this.setCreativeTab(CreativeTabs.MISC);
         this.setMaxStackSize(1);
         this.setMaxDamage(maxDurability);
-        this.setTranslationKey(itemName);
+        this.setUnlocalizedName(itemName);
         this.setRegistryName(itemName);
     }
 
@@ -42,7 +42,7 @@
     }
 
     public void addInformation(ItemStack stack, @Nullable World worldIn, List<String> tooltip, ITooltipFlag flagIn) {
-        tooltip.add(I18n.format("gui.shopkeeper." + this.getTranslationKey(), new Object[0]));
+        tooltip.add(I18n.format("gui.shopkeeper." + this.getUnlocalizedName(), new Object[0]));
     }
 
     public EnumSpecies getSpecies() {
