--- a/com/pixelmongenerations/common/item/ItemUltra.java
+++ b/com/pixelmongenerations/common/item/ItemUltra.java
@@ -16,7 +16,7 @@
         return damage;
     }
 
-    public String getTranslationKey(ItemStack stack) {
-        return super.getTranslationKey() + "_" + EnumDyeColor.byMetadata(this.getDamage(stack));
+    public String getTranslationKey(ItemStack stack) {
+        return super.getTranslationKey() + "_" + EnumDyeColor.byMetadata(this.getDamage(stack));
     }
 }
