--- a/com/pixelmongenerations/common/item/ItemZygardeCube.java
+++ b/com/pixelmongenerations/common/item/ItemZygardeCube.java
@@ -35,7 +35,7 @@
         ItemStack stack = playerIn.getHeldItem(hand);
         int damage = stack.getItemDamage();
         if (damage >= full) {
-            ChatHandler.sendChat(playerIn, "pixelmon.zygardecube.cellFull");
+            ChatHandler.sendChat(playerIn,  "pixelmon.zygardecube.cellFull");
         } else {
             ChatHandler.sendChat(playerIn, "pixelmon.zygardecube.cellAmount", damage);
         }
