--- a/com/pixelmongenerations/common/item/PixelmonItemBlock.java
+++ b/com/pixelmongenerations/common/item/PixelmonItemBlock.java
@@ -42,7 +42,7 @@
 
     public PixelmonItemBlock(Block block, String name) {
         super(block);
-        this.setTranslationKey(name);
+        this.setTranslationKey(name);
         this.setRegistryName(name);
     }
 
@@ -288,7 +288,7 @@
 
     @SideOnly(Side.CLIENT)
     public void addInformation(ItemStack stack, @Nullable World worldIn, List<String> tooltip, ITooltipFlag flagIn) {
-        String info = I18n.translateToLocal("gui.shopkeeper." + this.getTranslationKey());
+        String info = I18n.translateToLocal("gui.shopkeeper." + this.getTranslationKey());
         if (!this.hasHideFlag(stack) && !info.startsWith("gui.shopkeeper.")) {
             tooltip.add(GuiScreen.isShiftKeyDown() ? info : "Hold shift for more info.");
         }
