--- a/com/pixelmongenerations/common/item/tools/GenericHoe.java
+++ b/com/pixelmongenerations/common/item/tools/GenericHoe.java
@@ -26,12 +26,12 @@
 public class GenericHoe extends ItemHoe {
     public GenericHoe(ToolMaterial material, String itemName) {
         super(material);
-        this.setTranslationKey(itemName);
+        this.setTranslationKey(itemName);
         this.setRegistryName(itemName);
     }
 
-    public String getTranslationKey() {
-        return super.getTranslationKey().substring(5);
+    public String getTranslationKey() {
+        return super.getTranslationKey().substring(5);
     }
 
     public EnumActionResult onItemUse(EntityPlayer playerIn, World worldIn, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ) {
@@ -71,7 +71,7 @@
 
     @SideOnly(Side.CLIENT)
     public void addInformation(ItemStack stack, @Nullable World worldIn, List<String> tooltip, ITooltipFlag flagIn) {
-        String info = I18n.translateToLocal("gui.shopkeeper.item." + this.getTranslationKey());
+        String info = I18n.translateToLocal("gui.shopkeeper.item." + this.getTranslationKey());
         if (!this.hasHideFlag(stack) && !info.startsWith("gui.shopkeeper.")) {
             tooltip.add(GuiScreen.isShiftKeyDown() ? info : I18n.translateToLocal("pixelmon.item.tooltip"));
         }
