--- a/com/pixelmongenerations/client/gui/inventory/GuiInventoryPixelmonExtended.java
+++ b/com/pixelmongenerations/client/gui/inventory/GuiInventoryPixelmonExtended.java
@@ -25,7 +25,6 @@
         if (!this.inventory.isPokeInfoOpen()) {
             int i = this.guiLeft - 161;
             int j = this.guiTop;
-            int k = true;
             Collection<PotionEffect> collection = this.mc.player.getActivePotionEffects();
             if (!collection.isEmpty()) {
                 GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
@@ -46,14 +45,14 @@
                         this.drawTexturedModalRect(i, j, 0, 166, 140, 32);
                         if (potion.hasStatusIcon()) {
                             int i1 = potion.getStatusIconIndex();
-                            this.drawTexturedModalRect(i + 6, j + 7, 0 + i1 % 8 * 18, 198 + i1 / 8 * 18, 18, 18);
+                            this.drawTexturedModalRect(i + 6, j + 7, i1 % 8 * 18, 198 + i1 / 8 * 18, 18, 18);
                         }
 
                         potion.renderInventoryEffect(i, j, potioneffect, this.mc);
                         if (!potion.shouldRenderInvText(potioneffect)) {
                             j += l;
                         } else {
-                            String s1 = I18n.format(potion.getName(), new Object[0]);
+                            String s1 = I18n.format(potion.getName());
                             if (potioneffect.getAmplifier() == 1) {
                                 s1 = s1 + " " + I18n.format("enchantment.level.2", new Object[0]);
                             } else if (potioneffect.getAmplifier() == 2) {
