--- a/com/pixelmongenerations/core/command/GetBiomeData.java
+++ b/com/pixelmongenerations/core/command/GetBiomeData.java
@@ -37,7 +37,7 @@
             String strData = "";
 
             try {
-                strData = strData + biome.getRegistryName().getPath();
+                strData = strData + biome.getRegistryName().getResourcePath();
             } catch (Exception var14) {
                 strData = strData + "---";
             }
@@ -77,7 +77,7 @@
             strData = strData + ",";
 
             try {
-                strData = strData + biome.topBlock.getBlock().getTranslationKey();
+                strData = strData + biome.topBlock.getBlock().getUnlocalizedName();
             } catch (Exception var9) {
                 strData = strData + "---";
             }
