--- a/com/pixelmongenerations/common/block/tileEntities/TileEntityTrashcan.java
+++ b/com/pixelmongenerations/common/block/tileEntities/TileEntityTrashcan.java
@@ -19,7 +19,7 @@
     }
 
     public <T> T getCapability(Capability<T> capability, EnumFacing facing) {
-        return capability != CapabilityItemHandler.ITEM_HANDLER_CAPABILITY || facing != null && facing != EnumFacing.UP ? super.getCapability(capability, facing) : TileEntityTrashcan.VoidItemStackHandler.instance;
+        return capability != CapabilityItemHandler.ITEM_HANDLER_CAPABILITY || facing != null && facing != EnumFacing.UP ? super.getCapability(capability, facing) : (T) VoidItemStackHandler.instance;
     }
 
     public ITextComponent getDisplayName() {
