--- a/com/pixelmongenerations/common/spawning/PixelmonBiomeDictionary.java
+++ b/com/pixelmongenerations/common/spawning/PixelmonBiomeDictionary.java
@@ -116,8 +116,6 @@
 
     public static ArrayList<Integer> getBiomeIDsMatchingAllTags(ArrayList<EnumBiomeType> tags) {
         ArrayList<Integer> intBiomeIDs = new ArrayList();
-        int intCountMatches = false;
-
         for(int i = 0; i < biomeList.length; ++i) {
             int intCountMatches = 0;
             if (tags != null && !tags.isEmpty() && biomeList[i] != null && biomeList[i].typeList != null) {
@@ -141,7 +139,6 @@
 
     public static ArrayList<Integer> getBiomeIDsMatchingAllTagsExactly(ArrayList<EnumBiomeType> tags) {
         ArrayList<Integer> intBiomeIDs = new ArrayList();
-        int intCountMatches = false;
 
         for(int i = 0; i < biomeList.length; ++i) {
             int intCountMatches = 0;
