--- a/com/pixelmongenerations/common/battle/rules/clauses/SkyBattle.java
+++ b/com/pixelmongenerations/common/battle/rules/clauses/SkyBattle.java
@@ -20,7 +20,7 @@
     }
 
     public static boolean isMoveAllowed(Attack move) {
-        return !PROHIBITED_MOVES.contains(move.getAttackBase().getUnlocalizedName());
+        return !PROHIBITED_MOVES.contains(move.getAttackBase().getTranslationKey());
     }
 
     static {
