--- a/com/pixelmongenerations/core/util/PixelmonMethods.java
+++ b/com/pixelmongenerations/core/util/PixelmonMethods.java
@@ -28,9 +28,8 @@
             NBTTagCompound[] var4 = storage.getList();
             int var5 = var4.length;
 
-            for(int var6 = 0; var6 < var5; ++var6) {
-                NBTTagCompound nbt = var4[var6];
-                Optional pixelmonOptional;
+            for (NBTTagCompound nbt : var4) {
+                Optional<EntityPixelmon> pixelmonOptional;
                 if (nbt != null && (pixelmonOptional = storage.getAlreadyExists(getID(nbt), player.world)).isPresent()) {
                     list.add(pixelmonOptional.get());
                 }
