--- a/com/pixelmongenerations/core/util/PixelBlockSnapshot.java
+++ b/com/pixelmongenerations/core/util/PixelBlockSnapshot.java
@@ -189,8 +189,7 @@
     }
 
     public int hashCode() {
-        int hash = 7;
-        int hash = 73 * hash + this.pos.getX();
+        int hash = 73 * 7 + this.pos.getX();
         hash = 73 * hash + this.pos.getY();
         hash = 73 * hash + this.pos.getZ();
         hash = 73 * hash + this.meta;
